src/simplewallet/simplewallet.cpp

```diff
@@ -130,10 +130,25 @@ namespace
 
   const command_line::arg_descriptor< std::vector<std::string> > arg_command = {"command", ""};
 
+#ifdef HAVE_READLINE
+  class simplewallet_suspend_readline: public rdln::suspend_readline
+  {
+  public:
+    simplewallet_suspend_readline(): rdln::suspend_readline() { boost::lock_guard<boost::mutex> lock(mutex); in_password = true; }
```

Comment:
Why default `in_password` to `true`? Basically, as implemented, the question is "is readline suspended?" which doesn't really address the fact that the problem is actually in password_container:: read_from_tty AFAICT.

Reply:
I guess it should be "in_input" rather, yes. But we don't care since this gets mooted when you push the read_from_tty fix.